encoding/src/serde_byte_array.rs

```diff
@@ -68,13 +63,14 @@ mod tests {
 
     #[test]
     fn cannot_serialize_byte_array_overflow() {
+        let len = BYTE_ARRAY_MAX_LEN + 1;
         let bytes = ByteArray {
-            inner: vec![0; BYTE_ARRAY_MAX_LEN + 1],
+            inner: vec![0; len],
         };
 
         assert_eq!(
-            format!("{}", serde_cbor::to_vec(&bytes).err().unwrap()),
-            "Array exceed max length".to_string()
+            serde_cbor::to_vec(&bytes).err().unwrap().to_string(),
```

Comment:
Just a cosmetic question, wouldn't it be equivalent to not compare displays but only errors themselves? In other words, getting rid of `to_string()` in assertions?

Reply:
we had converted our Error types into the associated `Error` types with `ser::Error::custom` and `de::Error::custom` in serializing and deserializing(serde requires, see https://serde.rs/convert-error.html), and the error output in `serde_cbor::to_vec` converted the associated serde Error type into `serde_cbor::Error` which is `serde_cbor::Error::Message(string)` as result...so, we could not compare this `serde_cbor::Error` with our `SerdeError` but `to_string` can make this work

Reply:
@clearloop makes sense, thanks, I missed that. We could probably implement `From` for this type but that would be definitely an overkill, I'm fine with `String`. :+1: